sample_annotator/clients/submission_portal_client.py

```diff
@@ -0,0 +1,189 @@
+                            param_dic[key] = temp
+                            
+                        elif(key == "depth"):
+                            depth_lis = samp_dic[key].split("-")
+                            if len(depth_lis) != 2:
+                                temp = nmdc.QuantityValue(has_raw_value=samp_dic[key])
+                                param_dic[key] = temp
+                            else:
+                                temp = nmdc.QuantityValue(has_minimum_numeric_value = depth_lis[0],has_maximum_numeric_value = depth_lis[1])
+                                param_dic[key] = temp
+
+                        else:
+                            temp = nmdc.QuantityValue(has_raw_value=samp_dic[key])
+                            param_dic[key] = temp
+
+                    elif (key in float_params):
+                        if(key == 'elev'):
+                            temp = samp_dic[key].split()
+                            elev = float(temp[0])
+                            param_dic[key] = elev
+                        else:
+                            temp = float(samp_dic[key])
+                            param_dic[key] = temp
+
+                    elif (key in time_params):
+                        iso_date = parser.parse(samp_dic[key]).isoformat()
+                        temp = nmdc.TimestampValue(has_raw_value=iso_date)
+                        param_dic[key] = temp
+
+                    elif (key in geoloc_params):
+                        lat_lon_vals = samp_dic[key].split(',')
+                        temp = nmdc.GeolocationValue(has_raw_value=samp_dic[key],latitude=lat_lon_vals[0],longitude=lat_lon_vals[1])
+                        param_dic[key] = temp
+                    
+                    #if something shows up in the else statement it is missing from the mappings
+                    else:
+                        self.logger.info(key + " not found in any current param list")
+
+                if ('part_of'  not in param_dic.keys()):
+                    param_dic['part_of'] = '1000 soils'
+
+                if ('id' not in param_dic.keys()):
+                    param_dic['id'] = nmdc_ids[id_counter]
+                    id_counter = id_counter + 1
+
+                sample = nmdc.Biosample(**param_dic)
+
+                #For some rason water_content specifically was not being saved in the sample properly when unpacking the dictionary
+                if 'water_content' in param_dic.keys():
+                    sample['water_content'] = param_dic['water_content']
+                db.biosample_set.append(sample)
+            db_lis.append(db)
+        
+            
+        return db_lis
+            
+    
+    #dumps the given database to the given filepath
+    #params:
+    #database: an NMDC database object
+    #outfile_path: a string, filepath for desired output
+    def dump_db(self,database,outfile_path):
+        dumper.dump(element = database, to_file = outfile_path)
```

Comment:
Use `indent=4` as argument to `dump()` for better formatted JSON.

Reply:
I couldnt figure this one out, i didnt see an indent parameter for the function so this is unchanged at the moment